server/src/test/java/org/opensearch/index/store/remote/utils/TransferManagerTests.java

```diff
@@ -60,7 +60,7 @@ public void setUp() throws Exception {
         directory = new MMapDirectory(createTempDir(), SimpleFSLockFactory.INSTANCE);
         blobContainer = mock(BlobContainer.class);
         doAnswer(i -> new ByteArrayInputStream(createData())).when(blobContainer).readBlob(eq("blob"), anyLong(), anyLong());
-        transferManager = new TransferManager(blobContainer, fileCache);
+        transferManager = new TransferManager(blobContainer::readBlob, fileCache);
```

Comment:
All these `TransferManager` based tests are using the `blobContainer::readBlob`. We will need a variance of these tests with `RemoteDirectory` as well since it can be a `StreamReader` for composite directory use case.

Reply:
Ack. Will add it